src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/BlobContentId.cs

```diff
@@ -77,6 +77,24 @@ public static unsafe BlobContentId FromHash(byte[] hashCode)
             guidPtr[7] = (byte)((guidPtr[7] & 0x0f) | (4 << 4));
             guidPtr[8] = (byte)((guidPtr[8] & 0x3f) | (2 << 6));
 
+            // the code above assumed the host is little-endian; if this is not the case,
```

Comment:
Would it be better to just construct the Guid using `Guid (int a, short b, short c, byte d, byte e, byte f, byte g, byte h, byte i, byte j, byte k)` constructor instead of all the unsafe code and swapping bytes?

Reply:
Hmm, I initially tried to use a Guid constructor :-)   The first thing I tried was the `Guid (ReadOnlySpan<byte>)` constructor, which failed since the file is apparently also compiled against frameworks where there spans are not available.  Then I tried the `Guid (byte[])` constructor, which worked, but I was told this would be an unacceptable performance regression due to the extra allocation ...   That's when I went back to this solution that leaves the LE code path unaffected.

I can certainly try the separate-element constructor you point out, if that would be an acceptable change on the LE path.

Reply:
I think the separate-element constructor would be acceptable for both LE and BE paths.

Reply:
Patch updated accordingly.